pkg/controller/iotconfig/utils.go

```diff
@@ -62,12 +63,92 @@ func AppendEnvVarValue(container *corev1.Container, name string, value string) {
 	})
 }
 
+// block injection of sidecar variables, for containers not using jaeger
+func BlockTracingSidecarConfig(config *iotv1alpha1.IoTConfig, container *corev1.Container) {
+
+	if config.Spec.Tracing.Strategy.Sidecar != nil || config.Spec.Tracing.Strategy.DaemonSet != nil {
```

Comment:
I'm a bit confused about the DaemonSet, wouldn't this potentially add a tracing pod to all nodes on the cluster?

Reply:
Yes. The jaeger operator allows two modes: sidecar and daemonset.

I didn't want to impose either of them, so you need to align this configuration with your jaeger operator deployment.